rma/models/rma.py

```diff
@@ -0,0 +1,715 @@
+            # Use additional field helper function (for account extensions)
+            for line in refund.invoice_line_ids:
+                line._set_additional_fields(refund)
+            refund.message_post_with_view(
+                'mail.message_origin_link',
+                values={'self': refund, 'origin': self},
+                subtype_id=self.env.ref('mail.mt_note').id,
+            )
+            record.write({
+                'refund_id': refund.id,
+                'refund_line_id': refund_line.id,
+                'state': 'refunded',
+            })
+
+    @api.multi
+    def _prepare_refund(self):
+        self.ensure_one()
+        company_id = self.company_id.id
+        journal_id = self.env['account.invoice'].with_context(
+            company_id=company_id or self.env.user.company_id.id
+        ).default_get(['journal_id'])['journal_id']
+        if not journal_id:
+            raise UserError(_(
+                'Please define an accounting sales journal for this '
+                'company.'
+            ))
+        fiscal_position = self.partner_id.property_account_position_id
+        refund = self.env['account.invoice'].new({
+            'partner_id': self.partner_id.id,
+            'type': 'out_invoice',
+        })
+        # Get partner extra fields
+        refund._onchange_partner_id()
+        refund_vals = refund._convert_to_write(refund._cache)
+        refund_vals.update({
+            'origin': self.name,
+            'type': 'out_refund',
+            'account_id': self.partner_id.property_account_receivable_id.id,
+            'partner_shipping_id': self.partner_id.id,
+            'journal_id': journal_id,
+            'comment': "RMA note:\n{}".format(self.description),
+            'fiscal_position_id': fiscal_position.id,
+            'company_id': company_id,
+            'user_id': self.user_id.id,
+        })
+        return refund_vals
+
+    def _prepare_refund_line(self, invoice_id, qty):
+        self.ensure_one()
+        product = self.product_id.with_context(
+            force_company=self.company_id.id)
+        account = (product.property_account_income_id
+                   or product.categ_id.property_account_income_categ_id)
+        if not account and self.product_id:
+            raise UserError(
+                _('Please define income account for this product: '
+                  '"%s" (id:%d) - or for its category: "%s".') % (
+                    self.product_id.name,
+                    self.product_id.id,
+                    self.product_id.categ_id.name
+                )
+            )
+        fpos = self.partner_id.property_account_position_id
+        if fpos and account:
+            account = fpos.map_account(account)
+        return {
+            'invoice_id': invoice_id,
+            'name': self.product_id.name,
+            'origin': self.name,
+            'account_id': account.id,
+            'price_unit': self.product_id.lst_price,
```

Comment:
This should hook in a future module with the original sale order line price, if any. Maybe comment the method with something like:
"To be overriden in a third module with the proper origin values in case a sale order is linked to the original move"